console/src/components/App/Actions.js

```diff
@@ -46,6 +48,17 @@ const showNotification = ({
   };
 };
 
+export const requireAsyncGlobals = ({ dispatch }) => {
+  return (nextState, finalState, callback) => {
+    Promise.all([
+      dispatch(loadConsoleOpts()),
+      dispatch(fetchServerConfig()),
+    ]).finally(() => {
```

Comment:
What about just:
```suggestion
    ]).finally(callback)
```
?